### DISCUSSION TITLE
Identical naming of groups and subgroups

### DISCUSSION THREAD

**Discussion by JanEmde** [OP-TED/eForms-SDK#795] [Q&A] [answered]
We have noticed that individual groups and subgroups sometimes have the same name. For example there is a subgroup called "procedure" in the group "procedure". There are also groups with very similar names such as Result-1 and Result-2 or Tender-1 and Tender-2.  This causes irritation here and there. Are there plans to change these names in the future?

**Comment by YvesJo** [comment, marked as answer]
Hi,
I've converted the issue in discussion has it is not actually an SDK issue.
Some Groups are repeatable and need to be differentiated (e.g. Result-x, Tender-x ...)
As far as I know, there is no plan to change these names in the near future. If you have some expectations or suggestions, I invite you to share these with the helpdesk.
KR

**Comment by JanEmde** [comment]
Thanks for the answer. I get that some elements like "GR-LotResult-0" are repeatable. The problem is that for example the "GR-LotResult-0" contains the subgroups "GR-LotResult-1" and "GR-LotResult-2". These groups contain different BTs. 

When I repeat the element "GR-LotResult-0" i get the group "GR-LotResult-0" with then subgroups "GR-LotResult-1" and "GR-LotResult-2" AND the group "GR-LotResult-1" with the subgroups "GR-LotResult-1" and "GR-LotResult-2". That can be very irritating.

**Reply by YvesJo** [reply]
Hi,
I will try to clarify the case and will get back to you.

**Reply by TateGunning** [reply]
Not sure if machine is using it own mind or how to handle other than filling out the program paperwork (EU US IEC UN ISO ICO ASU - education) for credentials with certs downloaded, but currently in a housing transit with blotchy internet connectivity. Paperwork update confirmed everything on time and tracking. As for everything beyond understanding this process, generally speaking,, is there any specific thing needed to improve my internal system of understanding or modification? Please advise if this is relatable @YvesJo thanks,

**Reply by JanEmde** [reply]
Any news?

**Comment by rouschr** [comment]
Hello and happy new year,

> We have noticed that individual groups and subgroups sometimes have the same name

is this in your application, or in Enotices2, or in the editor demo?

One way to solve this would be to always do this when creating a new id:

- Find highest numbered id (abc-xxxx) in the notice for given prefix abc
- Increment number by 1

This would make the new id unique across the notice.
This also handles the problem of "graps" created by deletes, as it just looks for the highest number in the entire notice, you also have to take into account existing references (id-ref) and remove them when a referenced item is removed.

If this issue also exists on our side in one of our applications we will have to see if the NTD is problematic and should be modified or if the NTD is OK like this and it is the applications that should deal with this.

**Reply by JanEmde** [reply]
The problem is not about "creating a new id". 

There are fixed group names "Result 1", "Result 2", and "Tender 1", "Tender 2" and "Tender 3". This is confusing because there are also repeatable elements with the names "Result" and "Tender".

The group names should be clear and descriptive. For example "Result - Framework agreement".

![grafik](https://github.com/OP-TED/eForms-SDK/assets/144460224/d4de0806-df01-4a00-bf74-f06767d69705)

**Reply by YvesJo** [reply]
Hi,
Thanks for the clarification. We will update the labels . 
KR

**Reply by JanEmde** [reply]
When is the update planned for? SDK 1.11?

**Reply by YvesJo** [reply]
Hi,
This won't be in the SDK 1.11.0

**Reply by rouschr** [reply]
@YvesJo from what I understand we will try to put it in SDK 1.11 along with a few other label fixes
